tests/docker/Dockerfile

```diff
@@ -13,10 +13,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# The base image of openjdk:17 is typically oraclelinux:8-slim, which doesn't include apt-get. 
-# Therefore, use openjdk:17-buster instead.
-ARG jdk_version=openjdk:17-buster
-FROM $jdk_version AS build-native-image
+# We need to build the differnet versions of Java in same image, because the
+# zookeeper client 3.4.x can't run under JDK 17, so we need to use JDK 11 in 2.1 ~ 2.3 
+# kafka versions, see KAFKA-17888
+ARG os_version=debian:buster-20240423
+FROM $os_version AS build-native-image
+
+ARG java_base_dir=/opt/jdk
+ARG jdk_version_1=openjdk-11-jdk
```

Comment:
Could you please use `jdk_11_url` and `jdk_17_url` instead? It allows developers to test different JDK distributions easily.